src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -195,7 +205,7 @@ public GeneratorDriverTimingInfo GetTimingInfo()
             return new GeneratorDriverTimingInfo(_state.RunTime, generatorTimings);
         }
 
-        internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, DiagnosticBag? diagnosticsBag, CancellationToken cancellationToken = default)
+        internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, DiagnosticBag? diagnosticsBag, Func<GeneratorFilterContext, bool>? generatorFilter = null, CancellationToken cancellationToken = default)
```

Comment:
Rather than having the filter be optional here why not normalize it in the constructor to be non-null? Essentially somethnig like 

```csharp
generatorFilter ??= static _ => true;
```

Believe that should be amortized to a single alloc, if not could cache as a `static readonly` on the type

Reply:
Was purely to avoid the allocation, but yeah we can amortize it and it's probably simpler.

Reply:
Hmm, actually I think I'm going to leave this as-is. We don't take it in the ctor, it's passed on every call to run generators, we have to re-assign the empty one every run anyway, so it doesn't really make anything simpler.